beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1442,20 +1484,31 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
 
     /// Returns `HDiffBuffer` for the specified slot, or `HDiffBuffer` for the `ReplayFrom` slot if
     /// the diff for the specified slot is not stored.
-    fn load_hdiff_buffer_for_slot(&self, slot: Slot) -> Result<(Slot, HDiffBuffer), Error> {
+    fn load_hdiff_buffer_for_slot(
+        &self,
+        slot: Slot,
+        recursion: usize,
+    ) -> Result<(Slot, HDiffBuffer), Error> {
         if let Some(buffer) = self.diff_buffer_cache.lock().get(&slot) {
             debug!(
                 self.log,
                 "Hit diff buffer cache";
                 "slot" => slot
             );
+            metrics::inc_counter(&metrics::STORE_BEACON_DIFF_BUFFER_CACHE_HIT);
             return Ok((slot, buffer.clone()));
+        } else {
+            metrics::inc_counter(&metrics::STORE_BEACON_DIFF_BUFFER_CACHE_MISS);
         }
 
+        // Do not time recursive calls into load_hdiff_buffer_for_slot to not double count
```

Comment:
I'm not convinced how useful this `recursion` param is, because this will cause us just to time the top-level hdiff buffer request, which could be highly variable due to the number of recursive calls.

Reply:
Maybe we could time the whole call and then divide by `recursion` at the end? :thinking:

Reply:
I'm not super happy about the recursion param either. Counting the same time multiple times feels like it defeats the purpose of the metric tho. I would like to metric also how many recursions are necessary per hdiff load. Maybe we should move this metrics to the caller to avoid recursion?